src/fides/api/util/logger_context_utils.py

```diff
@@ -0,0 +1,93 @@
+from enum import Enum
+from typing import TYPE_CHECKING, Any, Dict, Optional
+
+from requests import PreparedRequest, Response
+from requests.exceptions import (  # pylint: disable=redefined-builtin
+    ConnectionError,
+    ConnectTimeout,
+    ReadTimeout,
+    SSLError,
+    TooManyRedirects,
+)
+
+if TYPE_CHECKING:
+    from fides.api.service.connectors.saas_connector import SaaSConnector
+
+from fides.config import CONFIG
+
+
+class ErrorGroup(Enum):
+    """A collection of user-friendly error labels to be used in contextualized logs."""
+
+    network_error = "Network/protocol error"
+    authentication_error = "Authentication error"
+    client_error = "Client-side error"
+    server_error = "Server-side error"
+
+
+def saas_connector_details(connector: "SaaSConnector") -> Dict[str, Any]:
+    """Maps the system and connection info details. Includes the collection and privacy request ID if available."""
+
+    details = {
+        "system_id": (
+            connector.configuration.system.id
+            if connector.configuration.system
+            else None
+        ),
+        "connection_key": connector.configuration.key,
+    }
+    if connector.current_collection_name:
+        details["collection"] = connector.current_collection_name
+    if connector.current_privacy_request:
+        details["privacy_request_id"] = connector.current_privacy_request.id
```

Comment:
nit: looking ahead to a _broader_ logging capability, can you think of any (simple) ways to make more generic/extensible the pieces that add the generic privacy request details to the log context/dictionary?

i.e., we'll probably want the `privacy_request_id` put on the logger context even when we don't have a `SaaSconnector` available to us (on non-SaaS DSRs). it'd be nice to ensure consistency/DRY code for adding that piece of context in all use cases.

all that being said, that may be an over-abstraction for now, if we're just looking to get the SaaS connector support, so totally fine deferring on this till later. just wanted to mention and keep eyes on what's ahead!

Reply:
one very rudimentary thing i can think of, at least for now, is just defining the `privacy_request_id` key as a module-level constant rather than a string defined inline. that's at least one measure we can take to help ensure consistency across contexts. we also could do the same for other keys that we anticipate will be reused across different context. 👍 


the reason i _key_ (pun intended) in on this is because consistency across logs is super important for being able to sift through the noise! so the closer we can get to a formal "schema" for the context we're adding (across the entire app), the better!